x-pack/plugins/ml/public/application/components/shared/pipeline_name_and_description.tsx

```diff
@@ -0,0 +1,116 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiFieldText, EuiFormRow, EuiText, EuiTextArea } from '@elastic/eui';
+
+// MlInferenceState
+interface Props {
+  handlePipelineConfigUpdate: (configUpdate: Partial<any>) => void;
+  pipelineNameError: string | undefined;
+  pipelineDescription: string;
+  pipelineName: string;
+}
+
+export const PipelineNameAndDescription: FC<Props> = ({
+  pipelineName,
+  pipelineNameError,
+  pipelineDescription,
+  handlePipelineConfigUpdate,
+}) => {
+  const handleConfigChange = (value: string, type: string) => {
+    handlePipelineConfigUpdate({ [type]: value });
+  };
+
+  return (
+    <>
+      <EuiFormRow
+        fullWidth
+        label={i18n.translate(
+          'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.configure.nameLabel',
+          {
+            defaultMessage: 'Name',
+          }
+        )}
+        helpText={
+          !pipelineNameError && (
+            <EuiText size="xs">
+              {i18n.translate(
+                'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.configure.name.helpText',
+                {
+                  defaultMessage:
+                    'Pipeline names are unique within a deployment and can only contain letters, numbers, underscores, and hyphens.',
+                }
+              )}
+            </EuiText>
+          )
+        }
+        error={pipelineNameError}
+        isInvalid={pipelineNameError !== undefined}
+      >
+        <EuiFieldText
+          data-test-subj="mlTrainedModelsInferencePipelineNameInput"
+          fullWidth
+          placeholder={i18n.translate(
+            'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.configure.namePlaceholder',
+            {
+              defaultMessage: 'Enter a unique name for this pipeline',
+            }
+          )}
+          value={pipelineName}
+          onChange={(e: React.ChangeEvent<HTMLInputElement>) =>
+            handleConfigChange(e.target.value, 'pipelineName')
+          }
+        />
+      </EuiFormRow>
+      {/* DESCRIPTION */}
+      <EuiFormRow
+        fullWidth
+        label={i18n.translate(
+          'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.configure.descriptionLabel',
+          {
+            defaultMessage: 'Description',
+          }
+        )}
+        helpText={
+          <EuiText size="xs">
+            {i18n.translate(
+              'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.configure.description.helpText',
+              {
+                defaultMessage: 'A description of what this pipeline does.',
```

Comment:
Alternatively, it can be a bit shorter.

```suggestion
                defaultMessage: 'A description of the pipeline.',
```

Reply:
Updated in 8e0c0ee7f9fc9d90db5838cb22de9c97fe1e7c46